### PR TITLE
Upgrade GitHub Actions to latest versions

### PR DESCRIPTION
**Pull request by sebastiaanspeck**

### CODE REVIEW

**Comment by StevenLooman** [comment]
I'm in doubt if we should upgrade to Java 17. While Java 11 only has [extended support](https://endoflife.date/oracle-jdk) at the moment, I expect there are companies which are still using an older version of SW and running Java 11.

**Comment by sebastiaanspeck** [comment]
> I'm in doubt if we should upgrade to Java 17. While Java 11 only has [extended support](https://endoflife.date/oracle-jdk) at the moment, I expect there are companies which are still using an older version of SW and running Java 11.

Sure, it is okay to stay on Java 11 for the time being.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -17,14 +17,14 @@ jobs:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
-      - name: Set up JDK 11
-        uses: actions/setup-java@v3
+      - name: Set up JDK 17
```
Comment (sebastiaanspeck) [inline .github/workflows/build.yml]:
```suggestion
      - name: Set up JDK 11
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -17,14 +17,14 @@ jobs:
...
+        uses: actions/setup-java@v4
         with:
-          java-version: 11
+          java-version: 17
```
Comment (sebastiaanspeck) [inline .github/workflows/build.yml]:
```suggestion
          java-version: 11
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -12,14 +12,14 @@ jobs:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
-      - name: Set up JDK 11
-        uses: actions/setup-java@v3
+      - name: Set up JDK 17
```
Comment (sebastiaanspeck) [inline .github/workflows/release.yml]:
```suggestion
      - name: Set up JDK 11
```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -12,14 +12,14 @@ jobs:
...
+        uses: actions/setup-java@v4
         with:
-          java-version: 11
+          java-version: 17
```
Comment (sebastiaanspeck) [inline .github/workflows/release.yml]:
```suggestion
          java-version: 11
```

**Comment by sebastiaanspeck** [comment]
You can also merge the PRs dependabot opened

**Comment by StevenLooman** [comment]
Changed my mind, upgraded to Java 17. We'll see what happens.

**Comment by StevenLooman** [comment]
Fixed via other PRs/changes. Thank you for creating this PR though!

**Pull request closed without merging**